### PR TITLE
#2 Chat command argument matching

### LINKED ISSUE #2: Chat command argument matching
Add chat command matching to plugin arguments

### PR DESCRIPTION
**Pull request by Thurston00**
Added functionality for commands to match a provided pattern from the plugin config to an arguments array passed to the trigger function

resolves #2

### CODE REVIEW

File: Nimda/Core/PluginContainer.php [hunk 1]
```diff
@@ -129,16 +129,41 @@ public function onMessage(Message $message)
...
 
+    /**
+     * @param $text
+     * @return Collection
+     */
     private function findPluginByCommand($text)
     {
         return $this->commands->filter(function ($plugin) use ($text){
-            return Str::startsWith($text, array_keys($plugin)[0]);
+            $commandShard = explode(' ', array_keys($plugin)[0])[0];
```
Comment (JABirchall) [inline Nimda/Core/PluginContainer.php:149]: Is there any better way to do this line?

File: Nimda/Core/PluginContainer.php [hunk 2]
```diff
@@ -129,16 +129,41 @@ public function onMessage(Message $message)
...
         })->collapse();
     }
+
+    private function parseArguments($message, $pattern)
+    {
+        $commandRegex = '/\{((?:(?!\d+,?\d+?)\w)+?)\}/';
+        $pattern = str_replace('/', '\/', $pattern);
+        $regex = '/^'.preg_replace($commandRegex, '(?<$1>.*)', $pattern).' ?/miu';
+
+        $regexMatched = (bool)preg_match($regex, $message, $matches);
```
Comment (JABirchall) [inline Nimda/Core/PluginContainer.php:160]: Can you explain what these regex lines do? looks like a lot.

Comment (Thurston00) [reply Nimda/Core/PluginContainer.php:160]: First of all, it replaces all instances of '/' to '\/' and matches it via the pattern for valid Regex. It then creates another Regex argument based on what command has been set. I.e if the command was !purge 5, in this instance, the preg_replace would set `(?<$1>.*)` to `(?<5>.*)` and that would be parsed and set as part of an array with the name "amount" as that's whats specified inside of the plugin itself.

`$regexMatched = (bool)preg_match($regex, $message, $matches);` returns a bool based on whether the Regex match is true or false.

File: Nimda/Core/PluginContainer.php (hunk 1 above)
Comment (Thurston00) [reply Nimda/Core/PluginContainer.php:149]: Not entirely sure. This method of doing it, is really bad though.

**Comment by JABirchall** [comment]
Looks good.

**Review by JABirchall — APPROVED** [review]

**Pull request merged**